src/main.rs

```diff
@@ -54,3 +77,59 @@ fn main() {
     let scores = sample_aggregate_scores::<5, rand::rngs::ThreadRng>(&mut rng, 20_000);
     println!("{scores}")
 }
+
+pub fn five_card(samples_number: u32){
+    let mut rng = rand::thread_rng();
+
+    println!("Sorted Deck:");
+    for id in 0..52 {
+        let card = Card { id };
+        if card.suit().id == 3 {
+            println!("  {}", card);
+        } else {
+            print!("  {}  ", card);
+        }
+    }
+
+    println!("");
+    let five_card_hand = Hand::<5>::draw(&mut rng);
+    println!("{five_card_hand}");
+    let hand_stats = HandStats::from(&five_card_hand);
+    println!("{hand_stats}");
+    let hand_score = HandScore::from(&hand_stats);
+    println!("{hand_score}");
+
+    // Now draw N random hands and check the stats!
+    println!("");
+    // Note: there is probably some way to deduce the type of the RNG here...
+    let scores = sample_aggregate_scores::<5, rand::rngs::ThreadRng>(&mut rng, samples_number);
+    println!("{scores}")
+}
+
+pub fn seven_card(samples_number: u32){
+    let mut rng = rand::thread_rng();
+
+    println!("Sorted Deck:");
+    for id in 0..52 {
+        let card = Card { id };
+        if card.suit().id == 3 {
+            println!("  {}", card);
+        } else {
+            print!("  {}  ", card);
+        }
+    }
+
+    println!("");
+    let seven_card_hand = Hand::<7>::draw(&mut rng);
+    println!("{seven_card_hand}");
+    let hand_stats = HandStats::from(&seven_card_hand);
+    println!("{hand_stats}");
+    let hand_score = HandScore::from(&hand_stats);
+    println!("{hand_score}");
+
+    // Now draw N random hands and check the stats!
+    println!("");
+    // Note: there is probably some way to deduce the type of the RNG here...
+    let scores = sample_aggregate_scores::<5, rand::rngs::ThreadRng>(&mut rng, samples_number);
+    println!("{scores}")
+}
```

Comment:
It looks like this function still bundles three different features that the binary can do:
- display the deck of cards (doesn't depend on the hand size)
- draw, score, and display a single hand
- compute statistics for a single hand size

Those should probably be three different execution modes, some of which are parameterized in terms of the hand size.